packages/aws_common/lib/src/logging/aws_logger.dart

```diff
@@ -63,6 +65,22 @@ class AWSLogger implements Closeable {
 
   final Logger _logger;
 
+  /// Parent of this logger in the logger hierarchy.
+  AWSLogger? get _parent {
+    return activeLoggers[_logger.parent?.fullName];
+  }
+
+  /// Children of this logger in the logger hierarchy.
+  final List<AWSLogger> _children = [];
+
+  Never get _pluginAlreadyRegistered {
```

Comment:
```suggestion
  Never _pluginAlreadyRegistered() {
```

Invoking a getter for side effects looks weird.

Reply:
Consider printing `T` as well